solver/src/solver/http_solver/model.rs

```diff
@@ -56,21 +58,23 @@ pub struct TokenInfoModel {
 
 #[derive(Debug, Serialize)]
 pub struct CostModel {
+    #[serde(with = "serde_with::rust::display_fromstr")]
     pub amount: u128,
-    pub token: String,
+    pub token: H160,
 }
 
 #[derive(Debug, Serialize)]
 pub struct FeeModel {
+    #[serde(with = "serde_with::rust::display_fromstr")]
     pub amount: u128,
```

Comment:
```suggestion
    #[serde(with = "u256_decimal")]
    pub weight: U256,
```

In theory order fees can be greater than `u128` (you can actually achieve this quite easily by having a token with extremely low value where the fee amount would be more than `2^128`).

Also note that this change does not affect the output JSON in any way - it just allows encoding a larger fee from the Rust side.

Reply:
I suppose we will want to change this pattern in all places where it occurs.